Added license and overview page (#439)
stackabletech/documentation: modules/ROOT/pages/product-information.adoc

= Product information

> This page contains contrete specifications about the Stackable Data Platform (SDP) as a product, which components are included, how they are supplied and which external dependencies exist that you as a customer need to take care of.

Reviewer: concrete

> This page contains contrete specifications about the Stackable Data Platform (SDP) as a product, which components are included, how they are supplied and which external dependencies exist that you as a customer need to take care of.

Reviewer: ```suggestion
This page contains concrete specifications about the Stackable Data Platform (SDP) as a product, which components are included, how they are supplied and which external dependencies exist that you as a customer need to take care of.
```

[#platform-components]
== Platform components

> The Stackable Platform is made up of multiple components. Operators for data products, Operators for additional functionality to facilitate easy integration between the different products. Auxilary software to interact with the platform, as well as software to ease the deployment of platform components.

Reviewer: and the data products themselves

> === Kubernetes Operators

Reviewer: ```suggestion
=== Data Products and Kubernetes Operators
```

> The main components of the SDP are Kubernetes Operators. Each Operator is supplied in a Docker <<containers, container>> image. For every Operator there is also a Helm Chart to facilitate installation via Helm. Operators for Products supported by the Platform deploy these products as <<containers, containers>> in Kubernetes.

Reviewer: The main components of the SDP are a set of open source data products. The SDP control plane consists of corresponding  Kubernetes Operators.

> The main components of the SDP are Kubernetes Operators. Each Operator is supplied in a Docker <<containers, container>> image. For every Operator there is also a Helm Chart to facilitate installation via Helm. Operators for Products supported by the Platform deploy these products as <<containers, containers>> in Kubernetes.

Reviewer: ```suggestion
The main components of the SDP are a set of open source data products. The SDP control plane consists of corresponding Kubernetes Operators. Each Operator is supplied in a Docker <<containers, container>> image. For every Operator there is also a Helm Chart to facilitate installation via Helm. Operators for Products supported by the Platform deploy these products as <<containers, containers>> in Kubernetes.
```

Supported products:

* xref:airflow:index.adoc[Apache Airflow]
* xref:druid:index.adoc[Apache Druid]
* xref:hbase:index.adoc[Apache HBase] (including xref:hbase:usage-guide/phoenix.adoc[Apache Phoenix])
* xref:hive:index.adoc[Apache Hive Metastore]
* xref:hdfs:index.adoc[Apache Hadoop HDFS]
* xref:kafka:index.adoc[Apache Kafka]
* xref:nifi:index.adoc[Apache Nifi]
* xref:spark-k8s:index.adoc[Apache Spark] (including xref:spark-k8s:usage-guide/history-server.adoc[Spark History Server])
* xref:superset:index.adoc[Apache Superset]
* xref:trino:index.adoc[Trino]
* xref:zookeeper:index.adoc[Apache Zookeeper]

> * xref:opa:index.adoc[Open Policy Agent]

Reviewer: I do not see OPA as a supported data product, but as a "helper" to enable Authorization. I would like to take it from the list of products.

Author: Ok I'll do it. I just grouped "product" vs "ours"

> * xref:opa:index.adoc[Open Policy Agent]

Reviewer: ```suggestion
```

> The product also includes the xref:commons-operator:index.adoc[Commons], xref:secret-operator:index.adoc[Secret] and xref:listener-operator:index.adoc[Listener] Operators.

Reviewer: add opa operator here instead

> The product also includes the xref:commons-operator:index.adoc[Commons], xref:secret-operator:index.adoc[Secret] and xref:listener-operator:index.adoc[Listener] Operators.

Reviewer: ```suggestion
The product also includes operators that provide additional functionality for managing and control SDP, i.e.the xref:commons-operator:index.adoc[Commons], xref:secret-operator:index.adoc[Secret] and xref:listener-operator:index.adoc[Listener] Operators.
```

The pages linked above also detail the use cases and features supported by each component. You can find additional information in the xref:release_notes.adoc[release notes]. Refer to the xref:operators:supported_versions.adoc[list of supported product versions] to find out which product versions are supported.

[#stackablectl]
=== stackablectl

xref:stackablectl::index.adoc[stackablectl] is a commandline utility that makes it easier to install and interact with stackable components.

== Supported installation methods

[#containers]
=== Operators and products

All operators are supplied in container images. The products are also deployed in container images.

> The docker images can be found here: https://repo.stackable.tech/#browse/browse:docker

Reviewer: ```suggestion
The docker images are available for download here: https://repo.stackable.tech/#browse/browse:docker
```

stackablectl is available for download, pre-built binaries are available on GitHub. The download link and installation steps are provided in the xref:stackablectl::installation.adoc[installation documentation].

== System requirements

=== Operators

Every Operator needs:

* 0.2 cores (i.e. i5 or similar)
* 256MB RAM

=== stackablectl

Hardware requirements and supported operating systems can be found in the xref:stackablectl::installation.adoc#system-requirements[system requirements section] of the stackablectl documentation.

=== Stackable Data Platform Open Source Products

The system requirements of each open source product depend on your specific use case. In the xref:stackablectl::demos/index.adoc[demos] you can find example use cases; every demo also has a _System requirements_ section which can provide a guideline for system sizing. Further example sizings can be found at https://ci.stackable.tech/job/<product>-operator-it-summary/ (where `<product>` is i.e. _druid_, _hbase_, etc.) in the `cluster-info.txt` file. These are the cluster sizes used for integration tests, and can be regarded as working cluster configurations.

== Prerequisites and required external components

> Required external components are components that are required for the platform or parts of it to operate, but are not part of the Stackable Data Platform. Note that Stackable does not distribute or offer support for the external components.

Reviewer: ```suggestion
Required external components are components that are required for the platform or parts of it to operate, but are not part of the Stackable Data Platform. Note that Stackable does neither distribute nor offer support for the external components.
```
